tchannel-core/src/main/java/com/uber/tchannel/handlers/ResponseRouter.java

```diff
@@ -87,8 +88,10 @@ protected void messageReceived(ChannelHandlerContext ctx, RawResponse rawRespons
         ResponsePromise<?> responsePromise = this.messageMap.remove(rawResponse.getId());
```

Comment:
So, the answer is: 'it depends'. If the Response is an 'error' response then yes. In the face of a fatal protocol error: 'no'.

There are a few scenarios under which we will fail:
- no response back: we don't gracefully handle this scenario
- Protocol `ErrorMessage`: we don't gracefully handle this scenario
- Response with `Error` response code: we successfully resolve the Promise.

I'll document these error edge-cases and see what we can do to cleanup our failure cases.